beefy-gadget/src/worker.rs

```diff
@@ -402,3 +384,133 @@ where
 
 	header.digest().convert_first(|l| l.try_to(id).and_then(filter))
 }
+
+/// Calculate the candidate block for the next BEEY vote.
+///
+/// Note, `number` is only used for better tracing output.
+fn vote_candidate<B>(
```

Comment:
Tried to use more restrictive trait bounds. Couldn't make it work in a way that is actually more succinct.

Reply:
Did you try using `BaseArithmetic` or `AtLeast32Bit`? The way it is right now requires that you always annotate the block type when calling the function since it cannot be deduced from the parameters.

Reply:
I tried `BaseArtihmetic`

Reply:
Pushed a change to address this.